File: frontend/src/app/model/apiResponse.ts
export class ApiResponse {
    message: string;
    status: number;
    httpStatus: string;
    timestamp: Date

    constructor(message: string, status: number, httpStatus: string, timestamp: Date) {
        this.message = message;
        this.status = status;
        this.httpStatus = httpStatus;
        this.timestamp = timestamp;
    }
}

File: common-module/src/main/java/ru/kraser/technical_helper/common_module/dto/api/ApiResponse.java
package ru.kraser.technical_helper.common_module.dto.api;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Builder;
import org.springframework.http.HttpStatus;

import java.time.LocalDateTime;

@Builder
public record ApiResponse(
    String message,
    int status,
    HttpStatus httpStatus,
    @JsonFormat(pattern = "MM-dd-yyyy HH:mm:ss")
    LocalDateTime timestamp,
    Object data
) {
}
